erpnext/selling/page/point_of_sale/point_of_sale.js

```diff
@@ -922,7 +939,25 @@ class POSCart {
 		this.$qty_total.find('.quantity-total').text(total_item_qty);
 		this.frm.set_value("pos_total_qty",total_item_qty);
 	}
-
+	make_order_type_options(){
+		this.order_type_field = frappe.ui.form.make_control({
+			df: {
+				fieldtype: 'Select',
+				label: 'Order Type',
+				fieldname: 'new_order_type',
+				options: this.frm.get_field("new_order_type").df.options,
+				reqd: 1,
+				default : this.frm.get_field("new_order_type").df.options[0],
```

Comment:
I think you can just use the default value set in the doc, as you've done below (`this.frm.doc.new_order_type`).

And remove line 959 as well.